Cannot read from URL on python 3

See e.g. https://travis-ci.org/kinverarity1/lasio/jobs/74413612

```bash
$ py.test --cov=lasio
============================= test session starts ==============================
platform linux2 -- Python 3.2.5 -- py-1.4.26 -- pytest-2.6.4
plugins: cov
collected 10 items 

lasio/test_open_file.py F....
lasio/test_read.py ....
lasio/test_wrapped.py .

=================================== FAILURES ===================================
________________________________ test_open_url _________________________________

    def test_open_url():
        l = read(
>           "https://raw.githubusercontent.com/kinverarity1/lasio"
            "/master/standards/examples/1.2/sample_curve_api.las")

lasio/test_open_file.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lasio/__init__.py:26: in read
    autodetect_encoding_chars=autodetect_encoding_chars)
lasio/las.py:156: in __init__
    autodetect_encoding_chars=autodetect_encoding_chars)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = {}
file_ref = 'https://raw.githubusercontent.com/kinverarity1/lasio/master/standards/examples/1.2/sample_curve_api.las'
encoding = None, autodetect_encoding = False, autodetect_encoding_chars = 20000

    def read(self, file_ref, encoding=None,
             autodetect_encoding=False, autodetect_encoding_chars=20000):
        '''Read a LAS file.
    
            Args:
              file_ref: either a filename, an open file object, or a string of
                a LAS file contents.
    
            Kwargs:
              encoding (str): character encoding to open file_ref with
              autodetect_encoding (bool): use chardet/ccharet to detect encoding
              autodetect_encoding_chars (int/None): number of chars to read from LAS
                file for auto-detection of encoding.
    
            '''
        f = open_file(file_ref, encoding=encoding,
                      autodetect_encoding=autodetect_encoding,
                      autodetect_encoding_chars=autodetect_encoding_chars)
    
        self._text = str(f.read())
        reader = Reader(self._text, version=1.2)
    
        self.version = reader.read_section('~V')
    
        # Set version
        try:
            reader.version = self.version['VERS'].value
        except KeyError:
>           raise KeyError("No key VERS in ~V section")
E           KeyError: 'No key VERS in ~V section'

lasio/las.py:186: KeyError
--------------- coverage: platform linux2, python 3.2.5-final-0 ----------------
Name                   Stmts   Miss  Cover
------------------------------------------
lasio/__init__             3      0   100%
lasio/las                389    145    63%
lasio/las2excel           53     53     0%
lasio/test_encoding        2      0   100%
lasio/test_open_file      17      0   100%
lasio/test_read           15      0   100%
lasio/test_wrapped         6      0   100%
------------------------------------------
TOTAL                    485    198    59%
====================== 1 failed, 9 passed in 0.64 seconds ======================

The command "py.test --cov=lasio" exited with 1.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ python:
     - "2.7"
     - "3.2"
     - "3.3"
+    - "3.4"
 before_install:
     - pip install pytest pytest-cov
     - pip install coveralls

--- a/lasio/las.py
+++ b/lasio/las.py
@@ -173,8 +173,22 @@ class LASFile(OrderedDictionary):
         f = open_file(file_ref, encoding=encoding,
                       autodetect_encoding=autodetect_encoding,
                       autodetect_encoding_chars=autodetect_encoding_chars)
-
-        self._text = str(f.read())
+        
+        if encoding is None:
+            encoding = "ascii"
+            logger.debug("Encoding not specified; set to %s" % encoding)
+        else:
+            logger.debug("Encoding=%s" % encoding)
+        
+        text = f.read()
+        logger.debug("file content has type %s" % type(text))
+        if isinstance(text, bytes):
+            self._text = text.decode(encoding)
+        elif isinstance(text, str):
+            self._text = text
+        else:
+            self._text = str(text)
+        
         reader = Reader(self._text, version=1.2)
 
         self.version = reader.read_section('~V')

--- a/notebooks/issues/issue #55.py
+++ b/notebooks/issues/issue #55.py
@@ -1,0 +1,7 @@
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+from lasio.test_open_file import test_open_url
+
+test_open_url()

